### PR TITLE
Add directorySelector

### LINKED ISSUE #3187: support for directorySelector
In Dymola there is a custom dialog annotation, Dialog.__Dymola_directorySelector which allows the user to select a directory.  There does not appear to be an equivalent in the Modelica Specification, or at least I could not figure out how to use loadSelector to do the same thing.

This is a useful feature could it be added?

### PR DESCRIPTION
**Pull request by HansOlsson**
As discussed at phone meeting the filter is not allowed.
Other design issues are discussed in #3187
Closes #3187

### CODE REVIEW

File: chapters/annotations.tex [hunk 1]
```diff
@@ -1328,6 +1329,10 @@ \section{Graphical User Interface}\label{annotations-for-the-graphical-user-inte
 \lstinline!caption! is a caption for display in the file dialog.
 \lstinline!loadSelector! is used to select an existing file for reading, whereas \lstinline!saveSelector! is used to define a file for writing.
 
+The presence of \lstinline!directorySelector! specifying \fmtannotationindex{Selector} suggests the use of a file dialog to select an existing directory.
+The \lstinline!filter! may not be used.
```
Comment (henrikt-ma) [inline chapters/annotations.tex:1333]: Maybe it wouldn't hurt to point out that it's OK to allow the dialog to be used for creating a directory on-the-fly?
```suggestion
The selected directory does not need to exist at the time of opening the dialog; it is allowed to let the dialog be used to create directory before selecting it.
The \lstinline!filter! may not be used.
```

Comment (henrikt-ma) [reply chapters/annotations.tex:1333]: (Unfortunately I needed to attach this suggestion to the line below in order to keep conversations apart.)

**Review by henrikt-ma — APPROVED** [review]
Looks good to me.

**Comment by henrikt-ma** [comment]
Oh, is automatic merging something we've adopted?  (I wasn't aware that my review would have this kind of immediate effect.)

**Comment by sjoelund** [comment]
@[HansOlsson](https://github.com/HansOlsson) HansOlsson enabled auto-merge [21 hours ago](https://github.com/modelica/ModelicaSpecification/pull/3207#event-6839163475)

**Comment by henrikt-ma** [comment]
OK.  Maybe it's reasonable for some really trivial stuff, but in general I think it would be better to give more time for collecting additional reviews before merging.

**Pull request merged**